network/f5/bigip_gtm_pool.py

```diff
@@ -0,0 +1,285 @@
+
+def get_all_pool_status(api):
+    try:
+        pool = api.GlobalLB.Pool.get_list()
+        return pool, api.GlobalLB.Pool.get_object_status(pool)
+    except Exception, e:
+        module.fail_json(msg="received exception: %s" % e)
+
+def get_pool_status(api, pool):
+    try:
+        return api.GlobalLB.Pool.get_object_status([pool])
+    except Exception, e:
+        module.fail_json(msg="received exception: %s" % e)
+
+def get_pool_state(api, pool):
+    state = api.GlobalLB.Pool.get_enabled_state([pool])
+    state = state[0].split('STATE_')[1].lower()
+    return state
+
+def get_pool_statistics(api, pool):
+    statistics = api.GlobalLB.Pool.get_statistics([pool])
+    return statistics
+
+def set_pool_state(api, pool, state):
+    state = "STATE_%s" % state.strip().upper()
+    api.GlobalLB.Pool.set_enabled_state([pool], [state])
+
+def pool_exists(api, pool):
+    # hack to determine if pool exists
+    result = False
+    try:
+        api.GlobalLB.Pool.get_object_status(pool_names=[pool])
+        result = True
+    except bigsuds.OperationFailed, e:
+        if "was not found" in str(e):
+            result = False
+        else:
+            # genuine exception
+            raise
+    return result
+
+def member_exists(api, pool, name, server):
+    # hack to determine if member exists
+    result = False
+    try:
+        members = [{'name': name, 'server': server}]
+        api.GlobalLB.Pool.get_member_object_status(pool_names=[pool], members=[members])
+        result = True
+    except bigsuds.OperationFailed, e:
+        if "was not found" in str(e):
+            result = False
+        else:
+            # genuine exception
+            raise
+    return result
+
+def add_pool_member(api, pool, name, server):
+    members = [{'name': name, 'server': server}]
+    api.GlobalLB.Pool.add_member_v2(pool_names=[pool], members=[members])
+
+def remove_pool_member(api, pool, name, server):
+    members = [{'name': name, 'server': server}]
+    api.GlobalLB.Pool.remove_member_v2(pool_names=[pool], members=[members])
+
+def add_pool(api, pool, lb_method):
+    lb_method = "LB_METHOD_%s" % lb_method.strip().upper()
+    # is order necessary?
+    api.GlobalLB.Pool.create_v2(pool_names=[pool], lb_methods=[lb_method],
+                               members=[[]], orders=[[]])
+def remove_pool(api, pool):
+    api.GlobalLB.Pool.delete_pool(pool_names=[pool])
+
+def main():
+    state_method_choices = ['state_disabled', 'state_enabled']
+    lb_method_choices = ['return_to_dns', 'null', 'round_robin',
+                                    'ratio', 'topology', 'static_persist', 'global_availability',
+                                    'vs_capacity', 'least_conn', 'lowest_rtt', 'lowest_hops',
+                                    'packet_rate', 'cpu', 'hit_ratio', 'qos', 'bps',
+                                    'drop_packet', 'explicit_ip', 'connection_rate', 'vs_score']
+
+    module = AnsibleModule(
```

Comment:
That's great. Thanks. I know we've had a number of PRs in the backlog for a while, together we can get that addressed :)